kotest-tests/kotest-tests-core/src/jvmTest/kotlin/com/sksamuel/kotest/core/runtime/parallel/SingleInstanceTest.kt

```diff
@@ -0,0 +1,291 @@
+package com.sksamuel.kotest.core.runtime.parallel
+
+import com.sksamuel.kotest.core.runtime.PersistentThreadLocal
+import io.kotest.assertions.assertSoftly
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.IsolationMode
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.ints.shouldBeExactly
+import io.kotest.matchers.ints.shouldBeGreaterThan
+import io.kotest.matchers.maps.shouldHaveSize
+import io.kotest.matchers.shouldBe
+import kotlinx.coroutines.delay
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.locks.ReentrantLock
+import kotlin.concurrent.getOrSet
+
+class SpecThreadSingleInstanceTest : FunSpec({
```

Comment:
We need a test for threads > 1 and threads < test count.

Reply:
I don't understand your idea. I already did test ` a test for threads > 1` All tests in this file `SingleInstanceTest.kt` are test with `threadsForSpec = 3` and there is example of ` threads < test count`. For example  `SpecThreadSingleInstanceTest` threads=3 and tests = 6

Reply:
Where is tests = 6 in this file ?

Reply:
SpecThreadSingleInstanceTest
Done